helix-view/src/modeline.rs

```diff
@@ -0,0 +1,258 @@
+use std::borrow::Cow;
+
+use helix_core::indent::IndentStyle;
+use helix_core::regex::Regex;
+use helix_core::{LineEnding, Rope};
+use once_cell::sync::Lazy;
+
+// 5 is the vim default
+const LINES_TO_CHECK: usize = 5;
+const LENGTH_TO_CHECK: usize = 256;
+
+static MODELINE_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"(?:\S+\s+)?(?:vi|vim|Vim|ex):\s*(?:set? )?(.*)(?::[^:]*)?").unwrap());
+
+#[derive(Default, Debug, Eq, PartialEq)]
+pub struct Modeline {
+    language: Option<String>,
+    indent_style: Option<IndentStyle>,
+    line_ending: Option<LineEnding>,
+}
+
+impl Modeline {
+    pub fn parse(text: &Rope) -> Self {
+        let mut modeline = Self::default();
+
+        for line in text.lines().take(LINES_TO_CHECK).chain(
+            text.lines_at(text.len_lines())
+                .reversed()
+                .take(LINES_TO_CHECK),
```

Comment:
Should this iterator be reversed again after the `take`? The lines at the end of the file would have the opposite merging behavior as the lines at the beginning otherwise

Reply:
fascinatingly enough no, the vim implementation also reads from the end of the file up